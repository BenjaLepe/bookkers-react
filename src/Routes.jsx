import React from 'react';
import { Route, Switch } from 'react-router-dom';
import NotFound from './views/NotFound';
import Login from './views/Login';
import BookDetail from './views/BookDetail';
import SignUp from './views/SignUp';
import Home from './views/Home';
import AuthContextProvider from './contexts/AuthContext';
import SessionProfile from './views/SessionProfile';
import SessionExpired from './views/SessionExpired';
import UserProfile from './views/UserProfile';
import CreateBook from './views/AddBook';
import EditUser from './views/EditUser';
import Navbar from './Components/Navbar';
import reviewReports from './views/Reports';
import EditBook from './views/EditBook';

export default function Routes() {
  return (
    <AuthContextProvider>
      <Navbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/sign-up" component={SignUp} />
        <Route exact path="/session-expired" component={SessionExpired} />
        <Route exact path="/session-profile" component={SessionProfile} />
        <Route exact path="/users/:id" component={UserProfile} />
        <Route exact path="/books/new" component={CreateBook} />
        <Route exact path="/books/:id" component={BookDetail} />
        <Route exact path="/edit-book/:id" component={EditBook} />
        <Route exact path="/edit-user/:id" component={EditUser} />
        <Route exact path="/books/:book_id/reviews/:review_id/reports" component={reviewReports} />
        <Route component={NotFound} />
      </Switch>
    </AuthContextProvider>
  );
}