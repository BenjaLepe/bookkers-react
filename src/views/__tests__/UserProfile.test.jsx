import React from 'react';
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom';
import Routes from '../../Routes';

const { act } = renderer;

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEntries={[ path ]}>
      <Routes />
    </MemoryRouter>
  );
}

const responseObj = {
  data: {
    data: {
      id: 2,
      username: 'responseUsername2',
      first_name: 'responseFirstName2',
      last_name: 'responseLastName2',
      email: 'response_test2@email.cl',
      description: 'responseDescription2',
    },
  },
};

const responseReview = {
  data: {
    data: null,
  },
};


jest.mock('../../hooks/calls', () => ({
  getUserRoute: () => responseObj,
  deleteUserRoute: () => null,
  getUserReviewsRoute: () => responseReview,
}));  

describe('User Profile', () => {
  describe('When there is an admin session', () => {
    const user = {
      access_token: 'TestToken',
      token_type: 'Bearer',
      user: {
        id: 1,
        username: 'testUserName',
        first_name: 'testFirstName',
        last_name: 'testLastName',
        email: 'test@email.cl',
        description: 'testDescription',
        is_admin: true,
      },
    };
    const sessionExp = new Date(
      new Date().getTime() + 1000 * 60 * 60 * 24, // One day in the future
    );
    const localStorageData = {
      user,
      sessionExp,
    };

    beforeEach(() => {
      global.Storage.prototype.getItem = jest.fn(
        (key) => JSON.stringify(localStorageData[key]),
      );
    });

    afterEach(() => {
      global.Storage.prototype.getItem.mockReset();
    });

    it('renders an user page', async () => {
      let testRenderer;

      act(() => {
        testRenderer = renderer.create(<TestRouter path="/users/2"/>);
      })

      await act(async () => {});

      const DOM = testRenderer.toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });
});