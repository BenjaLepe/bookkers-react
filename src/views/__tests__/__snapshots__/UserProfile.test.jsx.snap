// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile When there is an admin session renders an user page 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/session-profile"
          onClick={[Function]}
        >
          Profile
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/books/new"
          onClick={[Function]}
        >
          Add book
        </a>
        <button
          className="nav-button"
          onClick={[Function]}
          type="button"
        >
          Log out
        </button>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="container"
    >
      <div
        className="row info-header"
      >
        <div
          className="profile-pic-container"
        >
          <img
            alt="Profile picture"
          />
        </div>
        <div
          className="info-container"
        >
          <div>
            <span
              className="nombre-usuario"
            >
              responseFirstName2 responseLastName2 
            </span>
            <span
              className="nic-usuario"
            >
              @responseUsername2
               
              
            </span>
          </div>
          <div
            className="description-container"
          >
            <p>
              responseDescription2
            </p>
          </div>
          <div>
            <a
              href="/edit-user/2"
              onClick={[Function]}
            >
              <button
                type="button"
              >
                Edit User
              </button>
            </a>
            <button
              onClick={[Function]}
              type="button"
            >
              Ban User
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <h2>
          Reviews
        </h2>
        <p
          className="review-message"
        >
          No reviews
        </p>
      </div>
    </div>
  </div>,
]
`;
